### PR TITLE
fix: show non-existent plugin config

### LINKED ISSUE #841: :bomb: `Bug`: command `dtm show config -p harbor` print the wrong config sample
### What Happened?

- `./dtm show config -p harbor`

```sh
---
# core config
varFile: "" # If not empty, use the specified external variables config file
toolFile: "" # If not empty, use the specified external tools config file
state: # state config, backend can be local or s3
  backend: local
  options:
    stateFile: devstream.state

---
# plugins config
tools:
  - name: github-repo-scaffolding-golang
    instanceID: default
    options:
      owner: YOUR_GITHUB_USERNAME_CASE_SENSITIVE
      repo: go-webapp-devstream-demo
      branch: main
      image_repo: YOUR_DOCKER_USERNAME/go-webapp-devstream-demo
  - name: githubactions-golang
    instanceID: default
    dependsOn: ["github-repo-scaffolding-golang.default"]
    options:
      owner: YOUR_GITHUB_USERNAME_CASE_SENSITIVE
      repo: go-webapp-devstream-demo
      language:
        name: go
        version: "1.18"
      branch: main
      build:
        enable: True
      test:
        enable: True
        coverage:
          enable: True
      docker:
        enable: False

```

### How to Reproduce?

./dtm show config -p harbor

### Anything else

_No response_

### DevStream Version

latest

### PR DESCRIPTION
**Pull request by iyear**
## Pre-Checklist

Note: please complete **_ALL_** items in the following checklist.

- [x] I have read through the [CONTRIBUTING.md](https://github.com/devstream-io/devstream/blob/main/CONTRIBUTING.md) documentation.
- [x] My code has the necessary comments and documentation (if needed).
- [x] I have added relevant tests

## Description
I remove the default value of `template`. We can discuss better solution.
## Related Issues
close #841 

## New Behavior (screenshots if needed)
image

### CODE REVIEW

**Comment by daniel-hutao** [comment]
> In #795 , @IronCore864 thinks that templates are less suitable to be exported by command, but are suitable to be placed under `/examples`. The user gets the yaml file as a download.
> 
> Now two concurrent `quickstart.yaml` and `gitops.yaml` are preventing maintainability. If a solution can be discussed, I can fix it in the meantime.
> 
> @daniel-hutao @aFlyBird0

@iyear I've left a comment under #795. Let's have more discussions there.

**Comment by daniel-hutao** [comment]
For quickstart printing logic, my original design was "when `-p` does not exist, `-t` defaults to equal `quickstart` when `-t` is not set; when `-p` exists, `-t` does not take effect". 

We can follow this idea to fix it first. As for whether to delete the logic of `quickstart` configuration printing, we can wait until the end of the #795 discussion and implement it in a separate pr. @iyear

**Review by daniel-hutao — CHANGES_REQUESTED** [review]
see https://github.com/devstream-io/devstream/pull/847#issuecomment-1185273569

**Comment by iyear** [comment]
@daniel-hutao https://github.com/devstream-io/devstream/blob/2c8ba2d9e1cae2b0b1a1569a7d723dfb4b894173/internal/pkg/show/config/config.go#L22-L24

Then I have to remove this code.
Only one of the `-p` and `-t` args can be allowed to carry a default value.

**Comment by daniel-hutao** [comment]
I looked at the code and we need to prioritize support for `default config` printing, i.e. `dtm show config` can only print [internal/pkg/show/config/default.yaml](https://github.com/devstream-io/devstream/blob/d6875bbf7186cda670334f72f521ca314db29e38/internal/pkg/show/config/default.yaml). so we had to drop the default printing logic for `quickstart`. So, let's just keep the `-t quickstart` support.

@iyear Please also check if the documentation needs to be updated.

Releated issue: #597

File: internal/pkg/show/config/config.go [hunk 1]
```diff
@@ -6,14 +6,16 @@ import (
...
 //go:generate go run gen_embed_var.go
 func Show() error {
-	template := viper.GetString("template")
-	if template == "quickstart" {
-		fmt.Println(QuickStart)
-		return nil
-	} else if template == "gitops" {
-		fmt.Println(GitOps)
+	// at first, check is template parameter is set
+	if tmpl, ok := templates[viper.GetString("template")]; ok {
```
Comment (daniel-hutao) [inline internal/pkg/show/config/config.go:17]: What if a user type: `dtm show config -t abcd`?

It is possible that some exception input checks are needed here to inform the user what parameters are supported when the user enters an incorrect parameter.

**Review by daniel-hutao — CHANGES_REQUESTED** [review]

**Comment by iyear** [comment]
@daniel-hutao Is it possible to review and then merge in? I will start working on #795, otherwise it will cause a conflict.

**Pull request merged**